WordPress/gutenberg
# FormToggle: Covert component to TypeScript

## What?
Converts the `FormToggle` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `FormToggle` to TypeScript.
- Refactors the story to use TypeScript
- Refactors the tests to use `@testing-library/react` and snapshots

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `FormToggle` continues to function as expected
3. Test the various form toggles within the block/site editors in Gutenberg still work
